doc/format-run.md

```diff
@@ -199,7 +199,7 @@ The compact index type is designed to work with keys that are large
 cryptographic hashes, e.g. 32 bytes. In particular it requires:
 
 * keys must be uniformly distributed
-* keys must be at least 8 bytes (64bits), but can otherwise be variable length
+* keys should ideally be at least 8 bytes (64bits), but can otherwise be variable length
```

Comment:
I'm not totally sure it's true. If the keys are smaller then there's also fewer possible keys, since the keys are distinct.

For example, if the keys are 32bit. They're still distinct. So two adjacent keys (last one one page and first on next) will necessarily have distinct compact 64bit key prefixes. Indeed can't we _only_ get collisions for keys larger than 64bits?

Reply:
I changed the wording only because it's not a hard requirement from the code anymore -- no errors thrown if you have <8 byte keys

Also, keys don't have to necessarily be all of the same size. I could imagine a key like `data Key = Key1 Word32 | Key2 Word64` where the serialised key includes a single byte for the constructor, and then either a `Word32` or `Word64`. It's still a degenerate use case, I guess

Reply:
I'm fine with changing the wording back -- "keys must be uniformly distributed" is also not something we check

Reply:
We can't check they're uniformly distributed.

Suggestion:
```
* keys must be uniformly distributed;
* keys can be of variable length;
* keys less than 8 bytes (64bits) are padded with zeros (in LSB position).
```